src/NHibernate/Hql/Ast/ANTLR/Tree/EntityJoinFromElement.cs

```diff
@@ -15,12 +17,10 @@ public EntityJoinFromElement(FromClause fromClause, IQueryable entityPersister,
 			EntityType entityType = (EntityType) entityPersister.Type;
 			InitializeEntity(fromClause, entityPersister.EntityName, entityPersister, entityType, alias, tableAlias);
 
-			JoinSequence = new EntityJoinJoinSequenceImpl(
-				SessionFactoryHelper.Factory,
-				entityType,
-				entityPersister.TableName,
-				tableAlias,
-				joinType);
+			//Note: filters don't work with entity joins
```

Comment:
This comment is copied from `master`, where `EntityJoinJoinSequenceImpl` already become the same as here. Not sure if this comment is relevant or not.

Reply:
No it's not valid on master. It's been valid only for initial implementation and I just copy/pasted it. So it should go away with merge one way or another

Reply:
> filters don't work with entity joins

It's not valid because this PR is about working filters for entity joins  :)

Reply:
Ok, I've deleted comment.